src/ray/raylet/node_manager.cc

```diff
@@ -2929,6 +2929,7 @@ void NodeManager::ForwardTask(
         << "Spilling back to a node manager, but no GCS info found for node " << node_id;
     task.OnSpillback()(node_id, node_info->node_manager_address(),
                        node_info->node_manager_port());
+    on_error(ray::Status::IOError("Node not found when spilling back"), task);,
```

Comment:
This fix seems be correct. But I'm wondering if we need to make this more robust, so that developers shouldn't consider this   anymore, otherwise it will be broken often.

Reply:
There's many ways to make sure on_error called before return.
But I think it's better to rewrite forward logic.